forwards_paper.tex

```diff
@@ -462,14 +434,92 @@ \subsection*{Working with a tree sequence through tables}
 For some applications it is necessary to check that at every point
 there is only a \emph{single} tree.
 Checking this is more difficult, but is implemented in \msprime{}'s API.
-% \jda{TODO add function name}  PLR: not sure this is necessary.
 For efficiency, \msprime{} makes several other sortedness requirements on the tables,
 that are not necessarily satisfied by tables emitted by a forwards-time simulation.
 \msprime{}'s API includes tools to rectify this by first sorting % (using \texttt{sort\_tables})
 and then using the \texttt{simplify} algorithm described below, which works on sorted tables
 and is guaranteed to produce a valid, \msprime{}-ready tree sequence.
 
 
+
+%%%%%%
+\subsection*{The \msprime\ Tables API}
+
+The facilities for working with succinct tree sequences are implemented as part
+of the \msprime\ Python API, which provides a powerful platform for processing
+tree topology and mutation data. The new portions of \msprime{} that we discuss
+here are dedicated to tree sequence input and output using simple tables of
+data, so we refer to this as the `Tables API'. The four key tables (nodes,
+edges, sites and mutations) are described above.
+
+The Tables API is primarily designed to facilitate efficient interchange of
+data between programs or between different modules of the same program. We
+adopted a `columnar' design, i.e., where all the values for a
+particular column are stored in adjacent memory locations.
+There are many advantages to columnar storage---for example, since adjacent
+values in memory are from the same column, they tend to compress well,
+and suitable encodings can be chosen on a per-column basis~\citep{abadi2006integrating}.
+A particular advantage of this approach is that it enables very
+efficient copying of data, and in principle zero-copy data access
+(where a data consumer reads directly from the memory of a producer) is possible.
+Our implementation uses the NumPy C API~\citep{walt2011numpy} to efficiently copy
+data from Python into the low-level C library used to manipulate tree sequences.
```

Comment:
Note @molpopgen comment on earlier-merged #56 --- did you want this revised Kevin?

Reply:
see https://github.com/petrelharp/ftprime_ms/pull/60/commits/dfb3e1bcfb8a447d0c55d773ac1d8ef4f19e0401